here-naksha-storage-http/src/main/java/com/here/naksha/storage/http/POpToQuery.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (C) 2017-2023 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+package com.here.naksha.storage.http;
+
+import static com.here.naksha.lib.core.exceptions.UncheckedException.unchecked;
+import static com.here.naksha.lib.core.models.storage.POpType.*;
+
+import com.here.naksha.lib.core.models.storage.POp;
+import com.here.naksha.lib.core.models.storage.POpType;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.ArrayUtils;
+import org.jetbrains.annotations.NotNull;
+
+public class POpToQuery {
+
+  public static final String EQ_OPERATOR = "=";
+  public static final String NOT_EQ_OPERATOR = "!=";
+  public static final String NULL = ".null";
+  public static final String AND_DELIMITER = "&";
+
+  public static final String OR_DELIMITER = ",";
+  public static final String PATH_SEGMENT_DELIMITER = ".";
+
+  static String p0pToQuery(POp pOp) {
+    if (pOp.op() == AND) return and(pOp);
+    else return pOpToMultiValueComparison(pOp).resolve();
+  }
+
+  private static String and(POp pOp) {
+    assertHasAtLeastOneChildren(pOp);
+    return pOp.children().stream().map(POpToQuery::p0pToQuery).collect(Collectors.joining(AND_DELIMITER));
+  }
+
+  //
+  // MultiValueComparison operations
+  //
+  private static MultiValueComparison pOpToMultiValueComparison(POp pOp) {
```

Comment:
Consider removing these "section comments" - IMO they are not necessary and obscure this otherwise elegant class 🎩 

I am strongly against writing comments in source code - usually, this can be avoided by refactoring and naming stuff well. Following Uncle Bob:
> The purpose of comments is to explain code that cannot explain itself.

😄 This does not make any sense in this case as the code explains itself quite well here.

Couple of nice related blog posts about "when to comment" if you're interested:
- https://bpoplauschi.github.io/2021/01/20/Clean-Code-Comments-by-Uncle-Bob-part-2.html
- https://blog.cleancoder.com/uncle-bob/2017/02/23/NecessaryComments.html

Reply:
Done